llvm/lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -2073,6 +2073,16 @@ Instruction *InstCombinerImpl::visitPtrToInt(PtrToIntInst &CI) {
     return InsertElementInst::Create(Vec, NewCast, Index);
   }
 
+  // (ptrtoint (gep (inttoptr Base), Offset)) -> Base + Offset
+  Value *Base, *Offset;
+  if (match(SrcOp, m_OneUse(m_PtrAdd(m_OneUse(m_IntToPtr(m_Value(Base))),
+                                     m_Value(Offset)))) &&
+      Base->getType() == Ty && Offset->getType() == Ty) {
+    auto *NewOp = BinaryOperator::CreateAdd(Base, Offset);
+    NewOp->setHasNoUnsignedWrap(true);
```

Comment:
This nuw is incorrect. You need inbounds gep + non-negative offset for the add to be nuw.

Reply:
@YanWQ-monad Can you provide the alive proof?

Reply:
You can treat `inbounds` as `nsw + nuw`. See also https://discourse.llvm.org/t/rfc-add-nusw-and-nuw-flags-for-getelementptr/78672.

Reply:
It's now added in the description. Thanks.